qa/tasks/ceph.py

```diff
@@ -610,195 +600,258 @@ def cluster(ctx, config):
+        ctx.ceph = {}
+    ctx.ceph[cluster_name] = argparse.Namespace()
+
+    backup_ext = '.orig.teuthology'
+    devs_to_clean = {}
+    conf = configobj.ConfigObj()
+
+    if ctx.config.get('use_existing_cluster', False) is True:
+        log.info("'use_existing_cluster' is true; skipping cluster creation")
+
+        log.info("Backup ceph configuration file...")
+        remote = ctx.cluster.remotes.keys()[0]
+        remote.run(
             args=[
-                'sudo',
-                'install', '-d', '-m0777', '--', '/var/run/ceph',
+                'cp', conf_path, conf_path+backup_ext
             ],
-            wait=False,
+            wait=True,
+            check_status=True
         )
-    )
 
-    devs_to_clean = {}
-    remote_to_roles_to_devs = {}
-    osds = ctx.cluster.only(teuthology.is_type('osd', cluster_name))
-    for remote, roles_for_host in osds.remotes.items():
-        devs = teuthology.get_scratch_devices(remote)
-        roles_to_devs = assign_devs(
-            teuthology.cluster_roles_of_type(roles_for_host, 'osd', cluster_name), devs
-        )
-        devs_to_clean[remote] = []
-        log.info('osd dev map: {}'.format(roles_to_devs))
-        assert roles_to_devs, \
-            "remote {} has osd roles, but no osd devices were specified!".format(remote.hostname)
-        remote_to_roles_to_devs[remote] = roles_to_devs
-    log.info("remote_to_roles_to_devs: {}".format(remote_to_roles_to_devs))
-    for osd_role, dev_name in remote_to_roles_to_devs.items():
-        assert dev_name, "{} has no associated device!".format(osd_role)
-
-    log.info('Generating config...')
-    remotes_and_roles = ctx.cluster.remotes.items()
-    roles = [role_list for (remote, role_list) in remotes_and_roles]
-    ips = [host for (host, port) in
-           (remote.ssh.get_transport().getpeername() for (remote, role_list) in remotes_and_roles)]
-    mons = get_mons(
-        roles, ips, cluster_name,
-        mon_bind_msgr2=config.get('mon_bind_msgr2'),
-        mon_bind_addrvec=config.get('mon_bind_addrvec'),
+        log.info("get existing {cluster}.conf...")
+        proc = remote.run(
+                   args=[
+                       'cat', conf_path
+                   ],
+                   wait=True,
+                   check_status=False,
+                   stdout=StringIO(),
+                   stderr=StringIO()
+               )
+        if proc.exitstatus:
+            raise Exception("get existing cluster conf failed!")
+        conf = configobj.ConfigObj(proc.stdout, file_error=True)
+
+        log.info("set {cluster}.conf...".format(cluster=cluster_name))
+        if config.has_key('conf'):
+            for section, keys in config['conf'].iteritems():
+                for key, value in keys.iteritems():
+                    log.info("[%s] %s = %s" % (section, key, value))
+                    if section not in conf:
+                        conf[section] = {}
+                    conf[section][key] = value
```

Comment:
ACK, we're doing a merge.

Reply:
I don't understand, need I do some thing?

Reply:
nothing. @rzarzynski was just thinking aloud. explaining your change in his own words.

Reply:
Add the configuration in yaml to cluster.